providers/mlx5/qp.c

```diff
@@ -3771,8 +3775,10 @@ int mlx5_post_wq_recv(struct ibv_wq *ibwq, struct ibv_recv_wr *wr,
 			*bad_wr = wr;
 			goto out;
 		}
-
-		scat = get_wq_recv_wqe(rwq, ind);
+		if (rwq->is_mprq)
+			scat = get_wq_mprq_recv_wqe(rwq, ind);
+		else 
+			scat = get_wq_recv_wqe(rwq, ind);
 		sig = (struct mlx5_rwqe_sig *)scat;
 		if (unlikely(rwq->wq_sig)) {
```

Comment:
While this PR is open, I want to ask someone from the provides/mlx5 team if this change would interfere with the `memset` in this branch. Should I change the implementation advance the pointer `scat` after the memset?

Reply:
The functionality for Multi-Packet RQ was exposed over mlx5 DV API for the control path to be used later on by layers who may implement data path by themselves as of UCX/VMA, etc. and are fully familiar with the device specification. 

The data path flow is another full feature that may touch post_recv(), poll_cq(), etc.

As of that you can't just fix/adapt this small part, this will not work.

Please close this PR.

Reply:
I see, in that case would it be a good idea to post the `post_recv` function pointer in the ibv_wq structure to point to a function that explicitly errors to the user with the proper message rather than calling the current implementation that doesn't work. 
My concern is that (AFAIK) this not documented anywhere - that users should use rdma_core only for the control path and spin up their own implementation for the data path.

Reply:
As this is a DV API, we may expect its users to be familiar with the device specification/expected usage to some point.

For example the 'stride' properties from struct mlx5dv_striding_rq_init_attr are device specific ones, which to use them properly needs a close device knowledge.
 
As of that, we didn't set/replace a function pointer just for giving a non expected user an error message for such a DV scenario.

In case you really need a full striding RQ working solution please contact NVIDIA support for getting help/consider other layers as of UCX/VMA who may support that already.

Please close that PR as it can't work.

Thanks.

Reply:
Thanks for your response. I will close the PR.